feat: add pr_name config field

JustBeYou opened the pull request:
Close #1740

MarcoIeni commented on crates/release_plz_core/src/command/release_pr.rs:
```diff
@@ -415,6 +415,7 @@ async fn github_create_release_branch(
     release_branch: &str,
 ) -> anyhow::Result<()> {
     repository.push(release_branch)?;
+    // TODO @justbeyou would we like to create arguments for those hardcoded commit messages also?
```
Yes, I think it's what we need to do, right?

JustBeYou replied:
I think so, but I observed that the commit message and PR title had different sources for their messages and I was wondering if this should also be changed in this PR.

Anyway, I was just exploring the codebase and writing some comments for myself to familiarize with the codebase.

MarcoIeni replied:
I see :)

> I observed that the commit message and PR title had different sources for their messages and I was wondering if this should also be changed in this PR.

Oh, I see. Yeah, the PR name and the commit message should be the same. Feel free to do this refactoring in this pr or another one 👍

MarcoIeni commented on crates/release_plz_core/src/pr.rs:
```diff
@@ -64,20 +64,23 @@ fn pr_title(
 
+    // TODO @justbeyou seems like this piece of code is not covered by tests
+    // TODO @justbeyou we should probably add here the option to process a template if available
+    // otherwise we should keep the old behaviour
```
yes, exactly.

Regarding tests, there are some integration tests:
https://github.com/MarcoIeni/release-plz/blob/8d2bf351c29d6a7012444030441ca1d162af78b6/crates/release_plz/tests/all/main.rs#L1

JustBeYou commented:
@MarcoIeni, ready for review on this one 👀 

Short summary:
* did a little refactor of building Tera templates in `project.rs` and `tera.rs` as there was a little bit of annoying duplication
* implemented the PR name behavior and added tests
* synchronized the commit message in the release branch with the PR name

MarcoIeni commented on crates/release_plz_core/src/pr.rs:
```diff
@@ -64,7 +73,21 @@ fn pr_title(
+            first_version.as_str()
+        } else {
+            ""
+        };
```
sorry, I made a mistake, we shouldn't set the field if not present.
I.e. we shouldn't set empty string.

JustBeYou replied:
should we let them unset? what would be the behavior in this case? the literal `{{ version }}` string appear in the message? or should we error?

JustBeYou replied:
Failing seems the sanest way of handling things, as I think this feature will be used in very specific cases (looking at the original request). I added a commit to implement the change, let me know what you think

MarcoIeni replied:
Imo we should not set the fields. But i haven't looked into your changes yet.

MarcoIeni reviewed (approved):
Incredible contribution. Very high quality of the code. Many improvements to the existing codebase. Thanks 🙌

I hope you enjoyed working on release-plz and you will contribute in the future too. If you want to chat with me about anything, feel free to DM in my socials 👍

JustBeYou commented:
Thanks a lot for your support, it was a really awesome experience! I will take a look at some more issues in the following days and come back with some PRs.
